### PR TITLE
Add the calculation for transactions per second

### PR DESCRIPTION
**Pull request by smiasojed**
Add the TPS calculation to run.sh script

### CODE REVIEW

File: launch/run.sh [hunk 1]
```diff
@@ -103,8 +103,36 @@ if [ -n "${BINARIES_DIR}" ]; then
...
+do
+  num_extrinsics=$(echo "$line" | grep -o 'NumExtrinsics=[0-9]*' | cut -d'=' -f2) # Extract the number of extrinsics
+  total_extrinsics=$(echo "scale=0; $total_extrinsics + $num_extrinsics" | bc -l)
+  total_duration=$(echo "scale=2; $total_duration + 0.5" | bc -l )
+done <<< "$data"
+
+# Calculate the average number of extrinsics per second
+average_extrinsics_per_second=$(echo "scale=2; $total_extrinsics / $total_duration" | bc)
+
+echo "TPS=$average_extrinsics_per_second"
```
Comment (ascjones) [inline launch/run.sh:137]: We should provide some context as to what `TPS` means. In this case it is per second of execution time. The other measure is TPS including the block building time which is 12 or 6 secs with async backing. Possibly we should display both with a short explanation.

**Comment by xermicus** [comment]
Can we do this in `smart-bench` directly instead of using scripts?

File: src/stats.rs [hunk 2]
```diff
@@ -27,26 +26,59 @@ where
...
+                Ok(())
+            });
+            future::ready(result)
         })
+        .await?;
+    println!("\nSummary:");
+    println!("TPS - Transaction execution time per second, assuming a 0.5-second execution time per block");
+    println!(
+        "TPS: {}",
+        total_extrinsics as f64 / (total_blocks as f64 * 0.5)
```
Comment (ascjones) [inline src/stats.rs:81]: Would be good to display `total_extrinsics` and `total_blocks` too

File: src/stats.rs [hunk 3]
```diff
@@ -27,26 +26,59 @@ where
...
+    let mut total_blocks = 0u64;
+    println!();
+    block_info
+        .try_for_each(|block| {
+            println!("{}", block.stats);
+            let total_extrinsics_result = total_extrinsics
+                .checked_add(block.stats.num_extrinsics)
+                .ok_or_else(|| {
+                    color_eyre::Report::msg("Overflow occurred when calculating total extrinsics")
+                });
```
Comment (ascjones) [inline src/stats.rs:66]: It is unnecessary to use checked arithmetic here and below, can just use regular `+`, it's extremely unlikely to overflow `u64`.

**Review by ascjones — APPROVED** [review]

**Pull request merged**